ex1.py

```diff
@@ -0,0 +1,17 @@
+from collections import defaultdict
+
+
+def group_by(function, iterable):
+    #  this function groups values by the returned value
+    #  from the function that is provided and returns the dictionary
+    dictionary = defaultdict(list)  # creates an empty dictionary with empty list as value
+
+    key_and_value_tuple = []  # list of tuples
+
+    for value in iterable:
```

Comment:
Replace for loops -  
https://medium.com/python-pandemonium/never-write-for-loops-again-91a5a4c84baf
Replace for loops with more advanced python features